### ISSUE TITLE
Is there any gchange can be state.graded but the percentage is None?

### ISSUE THREAD

**Issue by dzhuang** [inducer/relate#457]
[This line](https://github.com/inducer/relate/blob/master/course/models.py#L1598) shows percentage can be `None`. If that's the case, [this line](https://github.com/inducer/relate/blob/master/course/models.py#L1753), and [this line](https://github.com/inducer/relate/blob/master/course/models.py#L1769) and [this line](https://github.com/inducer/relate/blob/master/course/models.py#L1778) will fail.

**Comment by inducer** [comment]
You're right. Those are broken. I think it's not inconceivable to see a grading opportunity with `max_points = 0`, which could lead to unavailable (`None`) percentage. So the lines you mention need to guard against that.

**Comment by dzhuang** [comment]
What's the use case is `max_points == 0`? Since we now have `is_optional_page`,  I think we can avoid that.
